### ISSUE TITLE
[buy] Update buy flow with fee breakdown design

### ISSUE THREAD

**Issue by jabby09** [KlimaDAO/klimadao#990]
This is related to the screen flow when buying carbon from a Seller Listing.

[Related design file](https://www.figma.com/file/5GewbdVaM1Q06CwHmF7d3F/%2Fprojects?node-id=11-59103&t=xRDvetMoatEVQakd-0).

![Screenshot 2023-03-29 at 5 07 41 PM](https://user-images.githubusercontent.com/103535195/228656653-9b5f4ff9-9ad3-4a4b-beec-4e27e3081c36.png)
![Screenshot 2023-03-29 at 5 11 10 PM](https://user-images.githubusercontent.com/103535195/228656670-01298e00-96bc-48a3-baef-0c40fdcd8746.png)

This is a useful page to reference to understand selective cost https://carbon.klimadao.finance/carbon-pricing

**Comment by jabby09** [comment]
@theprofessora can you please review the designs for this feature specifically and make sure it is aligning with any other /retire design changes you are making?

**Comment by theprofessora** [comment]
All updates have been made: https://www.figma.com/file/5GewbdVaM1Q06CwHmF7d3F?node-id=11:59103#402372817

**Comment by jabby09** [comment]
> All updates have been made: https://www.figma.com/file/5GewbdVaM1Q06CwHmF7d3F?node-id=11:59103#402372817

@goatsOnDrugs to be discussed

**Comment by jabby09** [comment]
@theprofessora let's make sure this flow / design matches what we just finalized for the buy from carbon pool flow.

https://www.figma.com/file/5GewbdVaM1Q06CwHmF7d3F?node-id=11:59103#402372817

**Comment by jabby09** [comment]
@ladytrekker adding this to the BUY from project details page epic.

This flow exists now but we need to update based on new designs and price details section plus other QoL changes to the form.

See design here: https://www.figma.com/file/5GewbdVaM1Q06CwHmF7d3F?node-id=11:59103#402372817

**Comment by ladytrekker** [comment]
@jabby09 Thanks. Will finish this first

**Comment by 0xAurelius** [comment]
Not all fees will apply to all listings

In particular, the `Liquidity Provider fee` and `Selective Retirement fee` only apply to "Pool" listings - the listing you shared was created by a user on Carbonmark at a fixed price, rather than being a Pool listing.

The Carbonmark API should expose whether a given listing is a pool or not since they display differently on the frontend: 
![image](https://github.com/KlimaDAO/klimadao/assets/91024694/7ed30b76-b012-47fd-acec-98a12ce1c42f)

For instance, [this project](https://www.carbonmark.com/projects/VCS-981-2014) (screenshotted above) has two listings from pools and one fixed-price seller listing

If the user is NOT retiring a pool token, then no fees are currently charged.

As for how to determine if a listing is from a pool or not via the Carbonmark API, and subsequently determine _which_ pool - I'm not sure, will defer to others on that

Does that help at all?

**Comment by ladytrekker** [comment]
@0xAurelius Thanks a lot!

This ticket here deals with a purchase from a seller listing.

If I get you right:
If a purchase is from a seller listing (not a pool project)
=> we can not render any fees

This means, the design above is meant to be for a pool purchase.
And this is another ticket: https://github.com/KlimaDAO/klimadao/issues/1127

@jabby09 @theprofessora Please double confirm. 
If we can leave out the fee, then this ticket is done.
See the screenshots from the opened pull request here: https://github.com/KlimaDAO/klimadao/pull/1143

**Comment by Atmosfearful** [comment]
@ladytrekker 
> How can I know the correct pool for a listing from a seller?

There is no "pool" for a listing from a seller, nor fees.


### First, here is a top-level view of the fees to make sure we are all on the same page
Fees depend on which payment token, which pool, and how many tonnes are being retired.
|                                                                        | SushiSwap | Toucan | C3     | KlimaDAO RA |
|-------------------------------------------------|-------------|---------|--------|----------------|
| Swap to pool token<br>(per swap)              | 0.3%          |              |        |             |
| Default Redeem from any pool                     | 0%             | 0%         |        |             |
| Selective Redeem from BCT                          |                   | 25%        |        |             |
| Selective Redeem from NCT                          |                  | 10%        |        |             |
| Selective Redeem from UBO                          |                  |              | 0.025% |             |
| Selective Redeem from NBO                          |                  |              | 0.025% |             |
| Retire your own project token                        |                  |              |        | no fee        |
| Redeem & retire pooled token<br>(% IN TONNES***) |  |        |        | 1%***          |

### Now breaking it down for each of our main journeys

#### Retire tco2/c3t you already own
No fees.
Why? The RA does not charge a fee if you are providing the project token to retire.

#### Buy tco2/c3t from 🏷️ seller listing
No fees.
Why? USDC goes directly to carbonmark contract. No swaps or redeems needed.

#### 'Retire Now' from 🏷️seller listing
No fees.
Why? Under the hood this is just a combination of both steps above (buy asset -> retire it)

#### 'Buy Now' from a 🤽pool listing.
No fees for the RA.
Sushi fees apply depending on # of swaps
Pool fees apply unless you are redeeming the default project.
Why?
Under the hood, the RA is doing a swap via sushi to get to the target pool token, then using the pool token to redeem the desired project from the pool. The RA does not charge for redeeming, it only charges for redeeming and retire.

#### 'Retire Now' from a 🤽pool listing.
Same as above, but with an additional 1% fee for the RA.

*** the 1% aggregator fee is calculated as 1% of the _tonnage being retired_.
For example, a default retirement of 1 tonne, paid for with BCT, adds a 0.01 BCT fee, for a total cost of 1.01 BCT
![image](https://github.com/KlimaDAO/klimadao/assets/88635679/b546d367-06f1-4ce2-95f0-b86744be36cf)
Then, this value is converted to USDC.

### Considering the above, there are a few challenges and open questions from my end
- Do we need to hardcode the # of swaps?
  - USDC -> MCO2 is 1 swap
  - USDC -> BCT is 2 swaps
  - When carbonmark adds support for other payment tokens, how many combinations will we need to support?
- Do we need to hardcode the current default projects for each pool?
  - These do not pay a pool redemption fee

**Comment by Atmosfearful** [comment]
@0xAurelius Clarification on this point
> 1% convenience fee charged on top of everything else (in carbon terms, so after the swap) by the retirement aggregator

Cujo is telling me it's just 1% of the tonnage being retired

**Comment by ladytrekker** [comment]
@Atmosfearful Thanks for detailed table.

I moved the information to the correct ticket here: https://github.com/KlimaDAO/klimadao/issues/1127
